File: 백준/Bronze/6122. Time Cards/Time Cards.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws Exception{
        BufferedReader I=new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st=new StringTokenizer(I.readLine());
        int n=Integer.parseInt(st.nextToken()),t=Integer.parseInt(st.nextToken());
        int[] sv=new int[n+1],time=new int[n+1];

        for(int i1=0;i1<t;i1++){
            String[] arr=I.readLine().split(" ");
            int tt=Integer.parseInt(arr[2])*60+Integer.parseInt(arr[3]);
            int me=Integer.parseInt(arr[0]);
            if(arr[1].equals("START")){
                sv[me]=tt;
            }else{
                time[me]+=tt-sv[me];
            }
        }
        StringBuilder ans=new StringBuilder();
        for(int i=1;i<n+1;i++){
            ans.append(time[i]/60).append(" ").append(time[i]%60).append('\n');
        }
        System.out.print(ans);
    }
}

File: 백준/Bronze/6122. Time Cards/Time Cards.py
import sys
def input(): return sys.stdin.readline().rstrip()

n,t=map(int,input().split())
save=[0]*(n+1)
time=[0]*(n+1)

for _ in range(t):
    ip=input().split()
    tt=int(ip[2])*60+int(ip[3])
    me=int(ip[0])
    if ip[1]=="START": save[me]=tt
    else: time[me]+=tt-save[me]

op=[]
for i in range(1,n+1):
    op.append(f"{time[i]//60} {time[i]%60}")

print('\n'.join(map(str,op)))